tests/deploy/api/test_e2e.py

```diff
@@ -6,8 +6,10 @@
 from ragna.deploy import Config
 from ragna.deploy._api import app
 from tests.deploy.utils import TestAssistant, authenticate_with_api
+from tests.utils import skip_on_windows
 
 
+@skip_on_windows
```

Comment:
What is the issue here? We shouldn't just skip our E2E test unless we are sure that this is testing issue on Windows.

Reply:
The [failure](https://github.com/Quansight/ragna/actions/runs/9891178695/job/27321052554?pr=438#step:4:169) looks real and serious:

```
Differing items:
  {'role': 'user'} != {'role': 'assistant'}
  {'id': 'a5a9ab44-2571-4dd1-889c-1ea8c61e6072'} != {'id': '0de4bf10-7e1c-4e54-8b8d-2aa760868335'}
  {'content': '?'} != {'content': "I'm a demo assistant and can be used to try Ragna's workflow.\nI will only mirror back my inputs. \n\nSo far I have received 1 messages.\n\nYour last prompt was:\n\n> ?\n\nThese are the sources I was given:\n\n- test.txt: ! "}
```

Reply:
Agreed, there's something funky going on with transaction processing on Windows but it's intermittent (sometimes the GH tests pass and sometimes they don't). I'll try to get to the bottom of this.

Reply:
Opened https://github.com/Quansight/ragna/issues/447